pkg/cloudprovider/vsphere/nodemanager_test.go

```diff
@@ -1883,3 +2105,23 @@ func TestExcludeLocalhostIPs(t *testing.T) {
 		t.Errorf("failure: expected ipAddr to equal fd00:100:64::1, but was %s", actual[1].ipAddr)
 	}
 }
+
+func guestInfoWithAddresses(addresses string) string {
```

Comment:
Maybe add another case for DHCP case? 
This is the guestInfo with DHCP machine
```
kubo@OrBVACRlEwEIY:~$ echo "Cmluc3RhbmNlLWlkOiAidGtnLW1nbXQtdmMtbWQtMC1uNWp4Zy01YzZiODc3NGN4a245NDYtNHdtN3ciCmxvY2FsLWhvc3RuYW1lOiAidGtnLW1nbXQtdmMtbWQtMC1uNWp4Zy01YzZiODc3NGN4a245NDYtNHdtN3ciCndhaXQtb24tbmV0d29yazoKICBpcHY0OiBmYWxzZQogIGlwdjY6IHRydWUKbmV0d29yazoKICB2ZXJzaW9uOiAyCiAgZXRoZXJuZXRzOgogICAgaWQwOgogICAgICBtYXRjaDoKICAgICAgICBtYWNhZGRyZXNzOiAiMDA6NTA6NTY6OTM6ZTI6NDIiCiAgICAgIHNldC1uYW1lOiAiZXRoMCIKICAgICAgd2FrZW9ubGFuOiB0cnVlCiAgICAgIGRoY3A0OiBmYWxzZQogICAgICBkaGNwNjogdHJ1ZQo=" |base64 -d

instance-id: "tkg-mgmt-vc-md-0-n5jxg-5c6b8774cxkn946-4wm7w"
local-hostname: "tkg-mgmt-vc-md-0-n5jxg-5c6b8774cxkn946-4wm7w"
wait-on-network:
  ipv4: false
  ipv6: true
network:
  version: 2
  ethernets:
    id0:
      match:
        macaddress: "00:50:56:93:e2:42"
      set-name: "eth0"
      wakeonlan: true
      dhcp4: false
      dhcp6: true
```

Reply:
We added a test for this case, thanks!